partner_communication_switzerland/migrations/10.0.2.4.6/post-migration.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import fields
+from datetime import datetime, timedelta
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    if not version:
+        return
+
+    sponsorships = env['recurring.contract'].search([
```

Comment:
You could replace this whole script by a SQL query which will be much faster :

1. Remove openupgrade.migrate decorator
2. `env` function parameter becomes `cr`
3.

```
cr.execute("""
update recurring_contract
set welcome_active_letter_sent=true
where activation_date <= %s
""", fields.Datetime.to_string(datetime.today() - timedelta(days=1)))
```